src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -160,8 +161,7 @@ public bool Equals(DiagnosticData? other)
             }
 
             return
-               DataLocation?.OriginalStartLine == other.DataLocation?.OriginalStartLine &&
-               DataLocation?.OriginalStartColumn == other.DataLocation?.OriginalStartColumn &&
+               DataLocation?.OriginalFileSpan?.StartLinePosition == other.DataLocation?.OriginalFileSpan?.StartLinePosition &&
```

Comment:
another form of equality.  why aren't we looking at the rest of teh span?  i don't know.  why aren't we looking at the mapped portino?  i don't know.